pkg/csi/service/wcp/controller.go

```diff
@@ -352,6 +354,100 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 	return resp, nil
 }
 
+// createFileVolume creates a file volume based on the CreateVolumeRequest.
+func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolumeRequest) (
+	*csi.CreateVolumeResponse, error) {
+	log := logger.GetLogger(ctx)
+	// ignore TopologyRequirement for file volume provisioning
+	if req.GetAccessibilityRequirements() != nil {
+		log.Info("Ignoring TopologyRequirement for file volume")
+	}
+
+	// Volume Size - Default is 10 GiB
+	volSizeBytes := int64(common.DefaultGbDiskSize * common.GbInBytes)
+	if req.GetCapacityRange() != nil && req.GetCapacityRange().RequiredBytes != 0 {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
+	volSizeMB := int64(common.RoundUpSize(volSizeBytes, common.MbInBytes))
+
+	var storagePolicyID string
+	for paramName := range req.Parameters {
+		param := strings.ToLower(paramName)
+		if param == common.AttributeStoragePolicyID {
+			storagePolicyID = req.Parameters[paramName]
+		}
+	}
+
+	var createVolumeSpec = common.CreateVolumeSpec{
+		CapacityMB:      volSizeMB,
+		Name:            req.Name,
+		StoragePolicyID: storagePolicyID,
+		ScParams:        &common.StorageClassParams{},
+		VolumeType:      common.FileVolumeType,
+	}
+
+	var volumeID string
+	var err error
+	if c.authMgr != nil {
```

Comment:
Do you need this check? If yes, what happens if this is nil?

Reply:
Good find. I don't we would need this. As we fail the file volume operation at the very start when auth check FSS is not enabled.